ext/storage_sources/s3_store/s3_storage_source.cpp

```diff
@@ -171,17 +146,70 @@ S3CustomizeFileSystem(WT_STORAGE_SOURCE *storageSource, WT_SESSION *session, con
 static int
 S3FileSystemTerminate(WT_FILE_SYSTEM *fileSystem, WT_SESSION *session)
 {
-    S3_FILE_SYSTEM *fs;
+    S3_FILE_SYSTEM *fs = (S3_FILE_SYSTEM *)fileSystem;
 
     UNUSED(session); /* unused */
 
-    fs = (S3_FILE_SYSTEM *)fileSystem;
     delete (fs->conn);
     free(fs);
 
     return (0);
 }
 
+/*
+ * S3DirectoryList --
+ *     Return a list of object names for the given location.
+ */
+static int
```

Comment:
We can name this to `S3ObjectList`. There is no concept of directories in an object store.